rest_framework/pagination.py

```diff
@@ -195,6 +195,7 @@ def paginate_queryset(self, queryset, request, view=None):
         Paginate a queryset if required, either returning a
         page object, or `None` if pagination is not configured for this view.
         """
+        self.request = request
```

Comment:
I like that you've moved these, so that the assignment is clearly consistent across all three cases.